curator-framework/src/main/java/org/apache/curator/framework/imps/RemoveWatchesBuilderImpl.java

```diff
@@ -0,0 +1,214 @@
+package org.apache.curator.framework.imps;

```

Comment:
Once the builder is committed via forPath() the watcher should also be removed from NamespaceWatcherMap. We should deprecate clearWatcherReferences() too.